robot/impl/resource_manager.go

```diff
@@ -410,11 +412,15 @@ func (manager *resourceManager) mergeResourceRPCAPIsWithRemote(r robot.Robot, ty
 }
 
 func (manager *resourceManager) closeResource(ctx context.Context, res resource.Resource) error {
-	allErrs := res.Close(ctx)
+	// Avoid hangs in Close/RemoveResource with resourceCloseTimeout.
+	closeCtx, cancel := context.WithTimeout(ctx, resourceCloseTimeout)
+	defer cancel()
```

Comment:
This code relies on builtin resources' `Close` methods actually respecting the context (see the `select` statement in `mockFake.Close`). If we're afraid those methods won't respect the context, we'll have to use a similar setup to what we have for resource (re)configuration timeouts with a goroutine that we potentially leak in the case of an unrespected context (cc @stuqdog). I opted for the more simple solution here.

Reply:
the broader ticket is tracked [here](https://viam.atlassian.net/browse/RSDK-6626)

Reply:
Left a `TODO` here.